### PR TITLE
ci: adds windows packaging and tests

### PR DESCRIPTION
**Pull request by mathetake**
the followup PR will configure the tag event trigger which creates a draft release and attach the artifacts.

### CODE REVIEW

File: packaging/msi/winget_manifest.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+Moniker: wazero
+Commands:
+  - wazero
+Tags:
+  - wazero
+  - webassembly
+  - wasm
+  - tetrate
+ShortDescription: wazero is the zero dependency WebAssembly runtime for Go developers
+Description: wazero is the zero dependency WebAssembly runtime for Go developers
```
Comment (mathetake) [inline packaging/msi/winget_manifest.sh:60]: I think I need your help on this @codefromthecrypt

Comment (codefromthecrypt) [reply packaging/msi/winget_manifest.sh:60]: indeed as this is a CLI we can distance from the programming part. maybe...

ShortDescription: wazero runs WebAssembly modules
Description: wazero is a command line utility to run WebAssembly modules. Specifically, this runs .wasm files compiled with WASI or Go, and has zero platform dependencies.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -45,10 +46,90 @@ jobs:
...
       - name: Make artifacts
-        run: | # Checks artifacts are buildable and versions can be correctly inserted with VERSION variable.
-          make dist VERSION=$GITHUB_SHA
-          tar xf dist/wazero_${GITHUB_SHA}_linux_amd64.tar.gz
-          ./wazero version | grep $GITHUB_SHA
+        env:
+          WINDOWS_CODESIGN_PASSWORD: ${{ secrets.WINDOWS_CODESIGN_PASSWORD }}
+        run: | # Checks artifacts are buildable.
+          make dist VERSION=${{ github.sha }}
+
```
Comment (codefromthecrypt) [inline .github/workflows/release.yaml:64]: CAP theorem strikes again ;)

File: packaging/msi/winget_manifest.sh [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+esac
+
+cat <<EOF
+# yaml-language-server: \$schema=https://aka.ms/winget-manifest.singleton.1.0.0.schema.json
+---
+PackageIdentifier: Tetrate.wazero
+PackageVersion: ${version}
+PackageName: wazero
+Publisher: Tetrate
+Copyright: Copyright 2021 Tetrate
```
Comment (codefromthecrypt) [inline packaging/msi/winget_manifest.sh:48]: maybe 2023 the things?

**Review by codefromthecrypt — APPROVED** [review]
great work. tricker than last time!

**Pull request merged**